core/trino-main/src/main/java/io/trino/sql/planner/SymbolAllocator.java

```diff
@@ -61,6 +61,9 @@ public Symbol newSymbol(String nameHint, Type type)
         requireNonNull(type, "type is null");
 
         nameHint = EXCLUDED_CHARACTERS.trimAndCollapseFrom(nameHint.toLowerCase(ENGLISH), '_');
+        if (nameHint.isEmpty()) {
```

Comment:
Add a check in Symbol constructor that provided name is not empty.

(if we allowed symbols named "", we should allow Variables named "")

Reply:
nit: Maybe move this just before line
```
Symbol symbol = new Symbol(type, nameHint);
```